### DISCUSSION TITLE
Skonstruowanie API w oparciu o REST vs CQRS vs CQS

### DISCUSSION THREAD

**Discussion by PrzemyslawKowalskiPK** [PLProjektKompetencyjny/PK_6IO1z_Projekt4#13] [General]
API w oparciu o REST jest typowym CRUD'em czyli w danym zasobie tworzy, pobiera, aktualizuje i usuwa dane.
Ścieżki po stronie są budowane w oparciu o schemat:
`/<resource>/[{id}]`
`/<resource>?param1=param1_val&param2=param2_val`

np.

`/users/1`
`/users?firstname=Andrew&lastname=Nowek`

W CQS'ie ścieżki w API są bardziej opisowe tj. definiują co poprzez dany endpoint jesteśmy w stanie uzyskać np.
`/reservations/cancel`
`/reservations/move`

W przypadku CQRS jest podobnie jak w CQS tzn. ścieżki w API są bardziej opisowe tj. definiują co poprzez dany endpoint jesteśmy w stanie uzyskać np.
`/reservations/cancel`
`/reservations/move`
Sam CQRS wymusza dodatkową architekturę m.in.:
- dodanie kolejki command'ów,
- dodanie Handler'ów do command'ów,
- utworzenie 2 baz danych jedna do odczytu (ReadModel) druga do zapisu,
- EDD (Event-Driven-Design).

Wydaje mi się, że CQRS namieszałby nam sporo tutaj więc raczej ta opcja by odpadała.
Należy rozważyć wybór REST'a lub CQS'a.

Wypodzcie się Panowie
@siemaEko @rukaSUKII @abienias97 @StanislawHornaGitHub 
Googlaj'cie i rozeznajcie się w terenie.
W przypadku CQRS'a niepoprawnie nakreśliłem, że to się ogranicza tylko do podzialu endpoint'ów. Dochodzi tam dużo logiki.

**Comment by rukaSUKII** [comment]
Ja bym osobiście szedł w REST - mniej kombinowania co z resztą wymieniłeś, że musimy obsłużyć wiele przypadków. Wystawienie REST na Flasku jest fajnie obsługiwane. Tutaj bym nie próbował szukać sobie utrudnień, bo możemy się za mocno zakręcić i potem będziemy musieli to cofać, gdzie finalnie wylądujemy w Rest-cie.

**Comment by abienias97** [comment]
Ja bym szedł w REST, raczej najpopularniejszy w technologiach webowych.
